kasa/smart/smartdevice.py

```diff
@@ -357,12 +357,25 @@ def alias(self) -> str | None:
     @property
     def time(self) -> datetime:
         """Return the time."""
-        if self._parent and Module.Time in self._parent.modules:
-            _timemod = self._parent.modules[Module.Time]
-        else:
-            _timemod = self.modules[Module.Time]
+        if (self._parent and (time_mod := self._parent.modules.get(Module.Time))) or (
+            time_mod := self.modules.get(Module.Time)
+        ):
+            return time_mod.time
 
-        return _timemod.time
+        # We have no device time, use current local time.
+        return datetime.now(timezone.utc).astimezone().replace(microsecond=0)
```

Comment:
I'd hope that we always have the time information, as it's probably not a good idea to use the host's time due to fluctiation..

Reply:
Agreed.  This logic was already there of sorts so I thought no harm in keeping it although none of the tests reached this line.